### PR TITLE
Fix result in case of requeue

### PR DESCRIPTION
**Pull request by rajdeepv**
Parallel cucumber merges the results by the order of worker index.

Consider the case: A the test failed on worker 4 and got rerun on worker 0 and passed. Still, the result is reported as failed because the merge starts from worker 0 upwards and the result of higher worker is considered

### CODE REVIEW

File: lib/parallel_cucumber/helper/cucumber/cucumber.rb [hunk 1]
```diff
@@ -43,8 +43,9 @@ def parse_json_report(json_report)
                        else
                          Status::UNKNOWN
                      end
-            [scenario, status]
-          end.to_h
+            report[scenario][:status] = status
```
Comment (fr0l) [inline lib/parallel_cucumber/helper/cucumber/cucumber.rb]: maybe inline variable ```status``` ?

File: lib/parallel_cucumber/worker_manager.rb [hunk 2]
```diff
@@ -67,7 +67,11 @@ def start_workers
         puts "Starting W#{index}"
         @workers["W#{index}"].start(env_for_worker(@options[:env_variables], index))
       end
-      @results.inject(:merge) # Returns hash of file:line to statuses + :worker-index to summary.
+      @results.inject do |seed, result|
+        seed.merge(result){|_key, oldval, newval|
+          (newval[:finish_time] > oldval[:finish_time]) ? newval : oldval
+        }
```
Comment (fr0l) [inline lib/parallel_cucumber/worker_manager.rb]: formatting. let's use ```do .. end``` for multiline. But as this one is quite straightforward let's just make it a one-liner but add spaces around curly brackets.

File: lib/parallel_cucumber/version.rb [hunk 3]
```diff
@@ -1,3 +1,3 @@
 module ParallelCucumber
-  VERSION = '0.2.18'.freeze
+  VERSION = '0.2.19.pre2'.freeze
```
Comment (fr0l) [inline lib/parallel_cucumber/version.rb]: why ```pre2``` ? maybe just ```0.2.19``` ?

Comment (bayandin) [reply lib/parallel_cucumber/version.rb]: I would recommend not to bump parallel_cucumber version in branches and to do it only in master

File: lib/parallel_cucumber/helper/cucumber/cucumber.rb (hunk 1 above)
Comment (rajdeepv) [reply lib/parallel_cucumber/helper/cucumber/cucumber.rb]: done

File: lib/parallel_cucumber/version.rb (hunk 3 above)
Comment (rajdeepv) [reply lib/parallel_cucumber/version.rb]: reverted

File: lib/parallel_cucumber/worker_manager.rb (hunk 2 above)
Comment (rajdeepv) [reply lib/parallel_cucumber/worker_manager.rb]: done

**Pull request merged**